File: Berger.Data/MsfaEntity/ELearning/QuestionSet.cs
using Berger.Data.Common;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Berger.Data.MsfaEntity.ELearning
{
    public class QuestionSet : AuditableEntity<int>
    {
        public string Title { get; set; }
        public int Level { get; set; }
        public int TotalMark { get; set; }
        public int PassMark { get; set; }
        public int TimeOutMinute { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public int ELearningDocumentId { get; set; }
        public ELearningDocument ELearningDocument { get; set; }
        //public int NeededCorrectAnswer { get; set; }
        public IList<QuestionSetCollection> QuestionSetCollections { get; set; }
        public IList<QuestionSetDepot> QuestionSetDepots { get; set; }
    }

    public class QuestionSetDepot : Entity<int>
    {
        public int QuestionSetId { get; set; }
        public QuestionSet QuestionSet { get; set; }
        public string Depot { get; set; }
    }
}

File: BergerMsfaApi/ClientApp/src/app/Shared/Entity/ELearning/questionSet.ts
import { QueryObject } from '../Common/query-object';
import { ELearningDocument } from './eLearningDocument';
import { QuestionSetCollection } from './questionSetCollection';

export class QuestionSet {
    id: number;
    title: string;
    level: number;
    totalMark: number;
    passMark: number;
    timeOutMinute: number;
    startDate: Date;
    endDate: Date;
    eLearningDocumentId: number;
    eLearningDocument: ELearningDocument;
    status: number;
    questionSetCollections: QuestionSetCollection[];
    depots: string[];

    statusText: string;
    
    constructor(init?: Partial<QuestionSet>) {
        Object.assign(this, init);
    }

    clear() {
        this.id = null;
        this.title = '';
        this.level = null;
        this.totalMark = null;
        this.passMark = null;
        this.timeOutMinute = null;
        this.startDate = null;
        this.endDate = null;
        this.eLearningDocumentId = null;
        this.status = null;
        this.questionSetCollections = [];
        this.depots = [];
        this.statusText = '';
    }
}

export class SaveQuestionSet {
    id: number;
    title: string;
    level: number;
    totalMark: number;
    passMark: number;
    timeOutMinute: number;
    startDate: Date;
    endDate: Date;
    eLearningDocumentId: number;
    status: number;
    questionSetCollections: QuestionSetCollection[];
    depots: string[];
    
    constructor(init?: Partial<SaveQuestionSet>) {
        Object.assign(this, init);
    }

    clear() {
        this.id = null;
        this.title = '';
        this.level = null;
        this.totalMark = null;
        this.passMark = null;
        this.timeOutMinute = null;
        this.startDate = null;
        this.endDate = null;
        this.eLearningDocumentId = null;
        this.status = null;
        this.questionSetCollections = [];
        this.depots = [];
    }
}

export class QuestionSetQuery extends QueryObject  {
    title: string;
    
    constructor(init?: Partial<QuestionSetQuery>) {
        super();
        Object.assign(this, init);
    }

    clear() {
        this.title = '';
    }
}